spp_pmt/models/custom_fields_ui.py

```diff
@@ -0,0 +1,20 @@
+# Part of OpenSPP. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class OpenSPPCustomFieldsUI(models.Model):
+    _inherit = "ir.model.fields"
+
+    field_weight = fields.Float("Default Weight", default=0)
+    with_weight = fields.Boolean(default=False)
+    area_ids = fields.One2many("spp.fields.area", "field_id", string="Areas")
+
```

Comment:
_sql_constraints = [("check_field_weight_pos", "CHECK(field_weight >= 0 AND field_weight <=1)", "Field weight should be between 0.0 and 1.0!")]